code/datums/components/material/material_container.dm

```diff
@@ -79,8 +79,10 @@
 
 	//can we insert into this container
 	if(!(mat_container_flags & MATCONTAINER_NO_INSERT))
+
 		//to insert stuff into the container
-		RegisterSignal(atom_target, COMSIG_ATOM_ITEM_INTERACTION, PROC_REF(on_item_insert))
+		var/signal = (mat_container_flags & MATCONTAINER_HIGH_PRIORITY) ? COMSIG_ATOM_ITEM_INTERACTION : COMSIG_ATOM_ATTACKBY
+		RegisterSignal(atom_target, signal, signal == COMSIG_ATOM_ATTACKBY ? PROC_REF(on_attack_by) : PROC_REF(on_item_insert))
```

Comment:
I think you missed the point of my first review - we should not be reverting anything to be using attackby for interactions. 

The end goal is to completely deprecate attackby for anything that isn't `I'm getting smacked by something and taking damage`

What I believe the problem is material containers just `return ITEM_INTERACT_SUCCESS` regardless of if the item is valid - It should either be non-blocking or more strict about when it returns that

Reply:
(or if that's not the problem, should mention what *is* the problem so we can work on)

Reply:
We have to block there is no other option.

Material container consumes(deletes) the item so we don't block other atoms down the line will be interacting with a deleted item.

Also machines like the flatpacker which overrides `item_interaction()` to accept the circuit board requires the material container to use `attackby()` else it would eat the circuit board and block this interaction

Reply:
Also we need the RPED to interact with the machine before the material container eats it so if we make the material container use `item_interaction()` signals it would block the rped  before it interacts with the machine so `attackby()` is also required here.

A long term solution is to have a proc that get's called on the "interacting atom" that is above `item_interaction()` for atoms like the material container. We don't have this problem with the "interacted atom" because we can override `base_item_interaction()` but we don't have a equivalent for the object we are clicking with

Reply:
Ok so after 5 seconds of deep thought I came with a 3 point solution 

1) add circuit board as a black listed item for material container so it won't eat and block the interaction a d allow the circuit board to be inserted 

2)Make RPED interact with atoms at the `tool_act()` level(because material container has taken up item_interaction() which is below `tool_act()`). It currently use yucky "pre_attack" so I can improve it's code as well.

I need to either 
A)Have atoms override tool_act so they interact with the RPED
B)Create a proc that gets called on the "interacting atom" from inside tool_act so the RPED can override and use it. (Perhaps something like `interface_with_atom()`).

I think I like the 2nd option more but yeah I'll get to work

Reply:
Ok after some thought i made some simple fixes